slack/ecs-deployment-failure/lambdas/main.rb

```diff
@@ -0,0 +1,98 @@
+require 'json'
+require 'net/http'
+
+FAILURE_EVENT_NAME = 'SERVICE_DEPLOYMENT_FAILED'.freeze
+
+def handler(event:, context:)
+  puts "[INFO] #{event}"
+  puts "[INFO] Context: #{context}"
+  event_name = event.fetch('detail').fetch('eventName')
+
+  return unless deployment_failure?(event_name)
+
+  resource_name = event.fetch('resources').join(',')
+  region = event.fetch('region')
+  reason = event.fetch('detail').fetch('reason')
+  _, cluster_name, service_name = resource_name.split('/')
+
+  response = post_to_slack(cluster_name, service_name, region, reason)
+  puts "[INFO] Response: #{response.body}"
+end
+
+def deployment_failure?(event_name)
+  event_name == FAILURE_EVENT_NAME
+end
+
+def post_to_slack(cluster_name, service_name, region, reason)
+  webhook_url = ENV['SLACK_WEBHOOK_URL']
+  uri = URI(webhook_url)
+  http = Net::HTTP.new(uri.host, uri.port)
+  http.use_ssl = true
+  request = Net::HTTP::Post.new(
+    uri.path,
+    { 'Content-Type' => 'application/json' }
+  )
+  request.body = payload(cluster_name, service_name, region, reason).to_json
+  http.request(request)
+end
+
+def payload(cluster_name, service_name, region, reason)
+  {
+    "blocks": [
+      {
+        "type": 'header',
+        "text": {
+          "type": 'plain_text',
+          "text": ':warning: Deployment Failure',
+          "emoji": true
+        }
+      },
+      {
+        "type": 'section',
+        "text": {
+          "type": 'mrkdwn',
+          "text": "*Cluster*: #{cluster_name}"
+        }
+      },
+      {
+        "type": 'section',
+        "text": {
+          "type": 'mrkdwn',
+          "text": "*Service*: #{service_name}"
+        }
+      },
+      {
+        "type": 'section',
+        "text": {
+          "type": 'mrkdwn',
+          "text": "*Region*: #{region}"
+        }
+      },
+      {
+        "type": 'section',
+        "text": {
+          "type": 'mrkdwn',
+          "text": "*Reason*: #{reason}"
+        }
+      }
+    ],
+    "attachments": [
+      {
+        "fallback": "Button to #{ecs_url(cluster_name, service_name, region)}",
```

Comment:
```suggestion
        "fallback": "Link to the AWS Console",
```

as per docs, `fallback` is plain text, so no need to print the entire URL here?

https://api.slack.com/messaging/composing/layouts#building-attachments